### DISCUSSION TITLE
Windows change default from pwsh to nushell

### DISCUSSION THREAD

**Discussion by batyrmuhadov** [LazyVim/LazyVim#3694] [Q&A]
Can anyone help me to setup nushell as the default, my config 
```
vim.o.shell = 'nu'
vim.o.shellcmdflag = '-c'
vim.o.shellquote = ""
vim.o.shellxquote = ""
```
when I am running codelens getting this error
`Error:   × Invalid literal
   ╭─[source:1:8]
 1 │ "cd "D:\Projects\Rustlang\temp" | cargo run --package temp --bin temp --all-features -- "  
   ·        ─────────────────────────────────────────┬────────────────────────────────────────  
   ·                                                 ╰── unrecognized escape after '\' in string
   ╰────

[Process exited 1]
`

**Comment by rubiin** [comment]
Maybe this would help you  https://github.com/nushell/nushell/issues/11821 , havent tried nushell myself

**Comment by kahojyun** [comment]
Rather than setting `vim.o.shell` to `nushell`, it might be better only using nushell in floating terminal. I override the default keymap with

```lua
-- lua/config/keymaps.lua
local shell = nil
if vim.fn.executable("nu") == 1 then
  shell = "nu"
end
local lazyterm = function()
  LazyVim.terminal(shell, { cwd = LazyVim.root() })
end
vim.keymap.set("n", "<leader>ft", lazyterm, { desc = "Terminal (Root Dir)" })
vim.keymap.set("n", "<leader>fT", function()
  LazyVim.terminal(shell)
end, { desc = "Terminal (cwd)" })
vim.keymap.set("n", "<c-/>", lazyterm, { desc = "Terminal (Root Dir)" })
vim.keymap.set("n", "<c-_>", lazyterm, { desc = "which_key_ignore" })
```
